internal/analytics/deferred.go

```diff
@@ -122,6 +122,9 @@ func sendDeferred(cfg Configurable, sender func(string, string, string, map[stri
 	if err != nil {
 		return errs.Wrap(err, "Could not load events on send")
 	}
+	if len(deferred) == 0 {
```

Comment:
Suggestion: It might be worth, instead, using `if len(deferred) > 0` and placing it around the cfg.Save below so that the "deferrerFilePath" is cleaned up.